File: android/app/src/main/java/com/fly/app/core/transfer/TransferItem.kt
package com.fly.app.core.transfer

sealed class TransferItem {
    abstract val id: String
    abstract val deviceName: String
    abstract val isIncoming: Boolean

    data class Text(
        override val id: String,
        override val deviceName: String,
        override val isIncoming: Boolean,
        val content: String
    ) : TransferItem()

    data class File(
        override val id: String,
        override val deviceName: String,
        override val isIncoming: Boolean,
        val name: String,
        val mime: String,
        val size: Long,
        val localPath: String? = null,
        val progress: Float = 0f,
        val totalChunks: Int = 0,
        val receivedChunks: Int = 0
    ) : TransferItem()

    data class LogcatStream(
        override val id: String,
        override val deviceName: String,
        override val isIncoming: Boolean,
        val lines: List<String> = emptyList()
    ) : TransferItem()
}

File: ios/Fly/Core/Transfer/TransferItem.swift
import Foundation

enum TransferItem: Identifiable {
    case text(id: String, deviceName: String, isIncoming: Bool, content: String)
    case file(id: String, deviceName: String, isIncoming: Bool, name: String, mime: String, size: Int64, localURL: URL?, progress: Double, totalChunks: Int, receivedChunks: Int)
    case logcatStream(id: String, deviceName: String, isIncoming: Bool, lines: [String])

    var id: String {
        switch self {
        case .text(let id, _, _, _): return id
        case .file(let id, _, _, _, _, _, _, _, _, _): return id
        case .logcatStream(let id, _, _, _): return id
        }
    }

    var isIncoming: Bool {
        switch self {
        case .text(_, _, let v, _): return v
        case .file(_, _, let v, _, _, _, _, _, _, _): return v
        case .logcatStream(_, _, let v, _): return v
        }
    }

    var deviceName: String {
        switch self {
        case .text(_, let v, _, _): return v
        case .file(_, let v, _, _, _, _, _, _, _, _): return v
        case .logcatStream(_, let v, _, _): return v
        }
    }
}
